client-app/shared/checkout/components/order-summary.vue

```diff
@@ -38,13 +32,13 @@
           <div v-if="cart.discounts && cart.discounts.length > 0 && discountsCollapsed">
             <ul v-for="(discount, index) in cart.discounts" :key="index" class="list-disc pl-5 text-gray-400">
               <li>
-                <div class="flex justify-between">
+                <div class="flex justify-between items-center">
                   <span class="text-sm">{{ discount?.description }}</span>
-                  <span>{{ cart.currency?.symbol }}{{ discount?.amount }}</span>
+                  <span>{{ cart.currency?.symbol }}{{ typeof discount?.amount === 'object' && discount?.amount !== null ? discount?.amount.amount : discount?.amount }}</span>
```

Comment:
Maybe it's better to create a computed prop instead. Template shouldn't content complex logic.